payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java

```diff
@@ -129,7 +129,9 @@ public PriorPaymentControlResult priorCall(final PaymentControlContext paymentCo
         final InternalCallContext internalContext = internalCallContextFactory.createInternalCallContext(paymentControlContext.getAccountId(), paymentControlContext);
         switch (transactionType) {
             case PURCHASE:
-                return getPluginPurchaseResult(paymentControlContext, pluginProperties, internalContext);
+                // if the Payment Method is still null, abort payment
+                return (paymentControlContext.getPaymentMethodId() != null) ? getPluginPurchaseResult(paymentControlContext, pluginProperties, internalContext) :
```

Comment:
@pierre 

Sure, I'll add the **log.warn** line.
Unfortunately the `invoiceId` is not present in the [PaymentControlContext](https://github.com/javier-gomez-hs/killbill/blob/13fd4001d59e59868186e2a306146d572be750dc/payment/src/main/java/org/killbill/billing/payment/core/sm/control/ControlPluginRunner.java#L260-L274).
Maybe I can use the **paymentId** instead and reply something like this:
 ```
log.warn("Payment was not triggered - Payment Method is null for PaymentId={}", paymentControlContext.getPaymentId());
```

Please let me know your thoughts.

Thanks!